### ISSUE TITLE
classic: slow performance

### ISSUE THREAD

**Issue by rbrewer123** [tomerfiliba/rpyc#379]
I really like the clarity and simplicity of code using classic rpyc.  My use case is for test orchestration.  However, the performance seems significantly slower than I expected.  

Here is my example:

```
from __future__ import print_function
import rpyc
import time

start = time.time()
conn = rpyc.classic.connect('192.168.56.101')
print('connect time: {}'.format(time.time() - start))
with conn.builtins.open('foobar.txt', 'wb') as fo:
    fo.write('hello there\n')
print('write file time {}'.format(time.time() - start))        
rpyc.utils.classic.download(conn, 'foobar.txt', 'foobar.txt')
print('download time {}'.format(time.time() - start))
```
This connects to `rpyc_classic.py` (in threaded mode) on a local VM (running on the same machine).

The output when I run it is as follows (note that the times are all relative to when the script started):
```
connect time: 1.93378710747
write file time 3.36589312553
download time 7.69390010834
```
I'm not at all concerned about the time to connect.  The time to write a file, and especially to download the same file seems excessively long.  

Is there something simple I'm doing wrong?  

My setup is:
rpyc 4.1.3
Python 2.7
Ubuntu 16.04 (client)
Windows 7 (rpyc_classic.py server)

When talking to another rpyc process running on the Ubuntu host, I see about a 2x speedup, but it still seems quite long.

**Comment by SofianJabal** [comment]
I'm also facing this issue what I moved to rpyc 4.1.4
on 4.1.2, it was working good

**Comment by SofianJabal** [comment]
I found the problematic commit,
It is 020f9fd562111688959ffdb2e10013c56ded34b0 ...

**Comment by rbrewer123** [comment]
@SofianJabal thanks for the tip.  I installed rpyc 4.1.2 and reran my test code above on the same setup.  The result is astonishingly 455x faster overall!

```
connect time: 0.0050060749054
write file time 0.00910401344299
download time 0.016902923584
```

**Comment by comrumino** [comment]
This looks to be duplicate of #382. Feel free to follow up in #382. I released 4.1.5 which should alleviate much of the slowness.
